addon/adapters/osf-adapter.js

```diff
@@ -10,14 +11,54 @@ export default DS.JSONAPIAdapter.extend(DataAdapterMixin, {
     authorizer: 'authorizer:osf-token',
     host: config.OSF.apiUrl,
     namespace: config.OSF.apiNamespace,
-
-    buildURL(modelName, id, snapshot, requestType, query) {  // jshint ignore:line
-        var url = this._super(...arguments);
+    buildURL(modelName, id, snapshot, requestType, query, dirtyRelationship) { // jshint ignore:line
```

Comment:
It seems that we're forcing the same save method to handle two separate kinds of requests; I think I'd almost prefer a separate `buildURL` method for relationships. (especially since the relationship functionality doesn't seem to require access to the parent method via `this._super`)

The other advantage of doing things that way is we wouldn't need to extend/alter the call signature of the function. (which could be fragile if ember added their own arguments in the future)